prov/efa/src/efa_domain.c

```diff
@@ -453,8 +453,223 @@ efa_domain_query_mr(struct fid_mr *mr, struct fi_efa_mr_attr *mr_attr)
 
 #endif /* HAVE_EFADV_QUERY_MR */
 
+#if HAVE_EFADV_QUERY_QP_WQS
+/**
+ * @brief Query EFA specific Queue Pair work queue attributes
+ *
+ * @param ep_fid  pointer to endpoint fid
+ * @param sq_attr pointer to send queue attributes
+ * @param rq_attr pointer to receive queue attributes
+ * @return 0 on success, negative integer on failure
+ */
+static int efa_domain_query_qp_wqs(struct fid_ep *ep_fid,
+				   struct fi_efa_wq_attr *sq_attr,
+				   struct fi_efa_wq_attr *rq_attr)
+{
+	struct efa_base_ep *base_ep;
+	struct efadv_wq_attr qp_sq_attr = {0};
+	struct efadv_wq_attr qp_rq_attr = {0};
+	int ret;
+
+	memset(sq_attr, 0, sizeof(*sq_attr));
+	memset(rq_attr, 0, sizeof(*rq_attr));
+
+	base_ep = container_of(ep_fid, struct efa_base_ep, util_ep.ep_fid);
+	ret = efadv_query_qp_wqs(base_ep->qp->ibv_qp, &qp_sq_attr, &qp_rq_attr, sizeof(qp_sq_attr));
+	if (ret) {
```

Comment:
Need to also check ```qp_sq_attr.comp_mask``` and ```qp_rq_attr.comp_mask``` are zero.

Reply:
For ABI compatibility, we may need some check in Libfabric's `fi_efa_wq_attr` in the future as well when we introduce more fields. New fields have to be introduced to the end of the structure, and we need to check Libfabric API version before we access the new fields from the sq_attr ptr from the application - because it may compile with an older Libfabric that do not allocate memory for these new fields